qp_qiime2/tests/test_qiime2.py

```diff
@@ -571,6 +602,49 @@ def test_beta_group_significance(self):
                 'qzv')]
         self.assertEqual(ainfo[0].files, exp)
 
+    def test_metrics(self):
+        # Test beta diversity metrics
+        beta_methods = q2div_plugin.methods['beta'].signature.parameters[
+            'metric'].qiime_type.predicate.choices
+        beta_alt_methods = q2div_plugin.methods[
+            'beta_phylogenetic_alt'].signature.parameters[
+                'metric'].qiime_type.predicate.choices
+        q2_metrics = beta_methods.union(beta_alt_methods)
+        qp_metrics = set(BETA_DIVERSITY_METRICS.values()).union(
+            STATE_UNIFRAC_METRICS.values()).difference(STATE_UNIFRAC_METRICS)
+        self.assertEqual(q2_metrics, qp_metrics)
```

Comment:
So if I parse the logic and the above conversation correctly, 

 1. we are hardcoding the dictionaries like `BETA_DIVERSITY_METRICS` so that we have human-friendly labels and will need to update them, but 
2. this test will check in with the installed Qiime2 version and complain to us if they've added or removed one of the option?

Reply:
That's correct! The test will fail if there is a metric present in the plugin not present in Qiime 2 and viceversa.

Reply:
This makes sense to me, although it sure would be nice if the human-readable element of the pair could be totally imported from Qiime2!

Reply:
Yeah... I think one of the main drawbacks is that the API itself is mainly tailored for programmers and CLI, which have different limitations than a GUI, which is probably why it is harder to provide those human readable elements from Q2 directly